src/dmd/parse.d

```diff
@@ -8919,14 +8919,19 @@ final class Parser(AST) : Lexer
         endtok = token.value == TOK.leftBracket ? TOK.rightBracket : TOK.rightParentheses;
 
         nextToken();
+
+        const errorsBefore = errorCount;
         while (token.value != endtok && token.value != TOK.endOfFile)
         {
             auto arg = parseAssignExp();
             arguments.push(arg);
-            if (token.value == endtok)
+            if (token.value == endtok || token.value != TOK.comma || errorCount != errorsBefore)
```

Comment:
I think you should not check the error count, as an error from an inner expression can be completely unrelated and might have been recovered from.